### PR TITLE
Adding a global Env to the extension API

### PR DESCRIPTION
**Pull request by sofianhn**
This will add a global Env variable that can contain information about VSCode. 

The Evn variable for now contains: 
- Version
- Machine Id. 
- Session Id. 
- Locale (to be implemented).

### CODE REVIEW

File: src/vs/vscode.d.ts [hunk 1]
```diff
@@ -9,10 +9,28 @@
 
 declare namespace vscode {
 
-	/**
-	 * The version of the editor.
-	 */
-	export var version: string;
```
Comment (jrieken) [inline src/vs/vscode.d.ts]: that is API breakage which we cannot do

File: src/vs/vscode.d.ts [hunk 2]
```diff
@@ -9,10 +9,28 @@
...
+	export const env: {
+		/**
+		 * The version of the editor.
+		 */
+		version: string;
+
+		/**
+		 * locale, like de-ch, en-us
+		 */
+		locale: string;
```
Comment (jrieken) [inline src/vs/vscode.d.ts]: @dbaeumer We are looking into adding some vscode environment information. What do thing is needed for localisation support. Is a `locale` string enough?

Comment (dbaeumer) [inline src/vs/vscode.d.ts]: @jrieken yes a string is enough. it is either pseudo or de-DE, ....

**Comment by SofianHn** [comment]
Addressed feedback, merging.

File: src/vs/workbench/api/node/extHost.api.impl.ts [hunk 3]
```diff
@@ -92,13 +95,23 @@ export class ExtHostAPIImplementation {
...
 		this._threadService = threadService;
+		this._telemetryService = telemetryService;
 		this._proxy = threadService.getRemotable(MainProcessVSCodeAPIHelper);
 
 		this.version = contextService.getConfiguration().env.version;
+		this._telemetryService.getTelemetryInfo().then((info) => {
+			this.env = {
+				machineId: info.machineId,
+				sessionId: info.sessionId,
+				locale: null
```
Comment (jrieken) [inline src/vs/workbench/api/node/extHost.api.impl.ts]: this is not good. either fill in or don't have it yet - i'll tackle

**Comment by dbaeumer** [comment]
@SofianHn you can get to the locale in the redener using: window.navigator.language

**Pull request merged**